paint/elections-phragmen/src/lib.rs

```diff
@@ -315,14 +312,81 @@ decl_module! {
 			let index = is_candidate.unwrap_err();
 
 			ensure!(!Self::is_member(&who), "member cannot re-submit candidacy");
+			ensure!(!Self::is_runner(&who), "runner cannot re-submit candidacy");
 
 			T::Currency::reserve(&who, T::CandidacyBond::get())
 				.map_err(|_| "candidate does not have enough funds")?;
 
 			<Candidates<T>>::mutate(|c| c.insert(index, who));
 		}
 
-		/// Remove a particular member from the set. This is effective immediately.
+		/// Denounce one's intention to be a candidate for the next election round. 3 potential
+		/// outcomes exist:
+		/// - `origin` is a candidate and not elected in any set. In this case, the bond is
+		///   unreserved, returned and origin is removed as a candidate.
+		/// - `origin` is a current runner up. In this case, the bond is taken and origin is removed
+		///   as a runner, consequently not being a candidate for the next round anymore. Indeed, a
+		///   runner up should generally avoid calling this function unless if they want a quick
+		///   exit out of the system. Otherwise, staying a runner and hoping to be elected down the
+		///   road makes more sense.
+		/// - `origin` is a current member. In this case, the bond is unreserved and origin is
+		///   removed as a member, consequently not being a candidate for the next round anymore.
+		///   Similar to [`remove_voter`], if replacement runners exists, they are immediately used.
+		///   Otherwise, a new election is started immediately.
+		// TODO: maybe don't re-run phragmen since it is expensive.
+		#[weight = SimpleDispatchInfo::FixedOperational(2_000_000)]
+		fn denounce_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+
+			// NOTE: this function attempts the 3 arms and fails if none are matched. Unlike other
```

Comment:
I am not really set on this and can change in case of any objections. The code would admittedly be more readable if we read all 3 sets first and do the checks.